service/controller/azuremachinepool/handler/nodepool/create_scale_workers.go

```diff
@@ -103,6 +103,13 @@ func (r *Resource) scaleUpWorkerVMSSTransition(ctx context.Context, obj interfac
 	desiredWorkerCount := int64(len(oldInstances) * 2)
 	r.Logger.Debugf(ctx, "The desired number of workers is: %d", desiredWorkerCount)
 
+	if desiredWorkerCount == 0 {
+		// The node pool is empty, the upgrade process is not finished.
```

Comment:
Comment seems to disagree with logged line.

Reply:
not == now. HOW DON'T YOU GET THAT :P

Reply:
fixed